extract-variables/README.md

```diff
@@ -0,0 +1,101 @@
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?> <root><city>San Jose</city><firstName>John</firstName><lastName>Doe</lastName><state>CA</state></root>
+```
+
+We use the  [Extract Variables](https://cloud.google.com/apigee/docs/api-platform/reference/policies/extract-variables-policy) policy as a first step to extract subsets of data from the XML response body. 
+Later on we leverage the [XML to JSON](https://cloud.google.com/apigee/docs/api-platform/reference/policies/xml-json-policy) policy to convert the XML response from the target server to JSON.
+We use again the  [Extract Variables](https://cloud.google.com/apigee/docs/api-platform/reference/policies/extract-variables-policy) policy to extract subsets of data from the now converted JSON string.
+Finally we use the [Assign Message](https://cloud.google.com/apigee/docs/api-platform/reference/policies/assign-message-policy?hl=en) policy to set the response headers with the extracted fields and other [flow variables](https://cloud.google.com/apigee/docs/api-platform/reference/variables-reference).
+
+## Prerequisites
+1. [Provision Apigee X](https://cloud.google.com/apigee/docs/api-platform/get-started/provisioning-intro)
+2. Configure [external access](https://cloud.google.com/apigee/docs/api-platform/get-started/configure-routing#external-access) for API traffic to your Apigee X instance
+3. Access to deploy proxies in Apigee
+4. Make sure the following tools are available in your terminal's $PATH (Cloud Shell has these preconfigured)
+    * [gcloud SDK](https://cloud.google.com/sdk/docs/install)
+    * unzip
+    * curl
+    * jq
+    * npm
+# (QuickStart) Setup using CloudShell
+
+Use the following GCP CloudShell tutorial, and follow the instructions.
+
+[![Open in Cloud Shell](https://gstatic.com/cloudssh/images/open-btn.png)](https://ssh.cloud.google.com/cloudshell/open?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/apigee-samples&cloudshell_git_branch=main&cloudshell_workspace=.&cloudshell_tutorial=extract-variables/docs/cloudshell-tutorial.md)
+
+## Setup instructions
+
+1. Clone the `apigee-samples` repo, and switch the `extract-variables` directory
+
+
+```bash
+git clone https://github.com/GoogleCloudPlatform/apigee-samples.git
+cd apigee-samples/extract-variables
+```
+
+2. Edit `env.sh` and configure the following variables:
+
+* `PROJECT` the project where your Apigee organization is located
+* `APIGEE_HOST` the externally reachable hostname of the Apigee environment group that contains APIGEE_ENV
+* `APIGEE_ENV` the Apigee environment where the demo resources should be created
+
+Now source the `env.sh` file
+
+```bash
+source ./env.sh
+```
+
+3. Deploy Apigee API proxies:
+
+```bash
+./deploy.sh
+```
+
+## Example Requests
+To manually test the proxy, make requests using curl:
+
+```
+curl https://$APIGEE_HOST/v1/samples/extract-variables
```

Comment:
small suggestion: add the `-v` flag so the user can see the headers returned in the response